src/kOS.Safe/Compilation/KS/Compiler.cs

```diff
@@ -413,76 +413,85 @@ private void PreProcessChildNodes(ParseNode node)
         private void PreProcessOnStatement(ParseNode node)
         {
             NodeStartHousekeeping(node);
+            nextBraceIsFunction = true; // triggers aren't really functions but act like it a lot.
+
```

Comment:
This change was to support having the ``return`` keyword work correctly from within the trigger body.